app/src/userpages/modules/userPageStreams/actions.js

```diff
@@ -396,32 +404,32 @@ export const uploadCsvFile = (id: StreamId, file: File) => (dispatch: Function)
         },
         withCredentials: true,
     })
-        .then(() => {
-            dispatch(uploadCsvFileSuccess())
+        .then((response) => {
+            if (response.data.schema.timestampColumnIndex == null) {
+                dispatch(uploadCsvFileUnknownSchema(id, response.data.fileId, response.data.schema))
+                throw new Error('Could not parse timestamp column!')
```

Comment:
This marks the upload as failed as it goes to the `catch` block but the upload still opens the confirm modal.